FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -500,70 +506,88 @@ - (nullable instancetype)initWithAPIKey:(NSString *)APIKey
     [GULSceneDelegateSwizzler proxyOriginalSceneDelegate];
 #endif  // TARGET_OS_IOS
 
-    // Continue with the rest of initialization in the work thread.
-    __weak FIRAuth *weakSelf = self;
-    dispatch_async(FIRAuthGlobalWorkQueue(), ^{
-      // Load current user from Keychain.
-      FIRAuth *strongSelf = weakSelf;
-      if (!strongSelf) {
-        return;
-      }
-      NSString *keychainServiceName =
-          [FIRAuth keychainServiceNameForAppName:strongSelf->_firebaseAppName];
-      if (keychainServiceName) {
-        strongSelf->_keychainServices =
-            [[FIRAuthKeychainServices alloc] initWithService:keychainServiceName];
-        strongSelf.storedUserManager =
-            [[FIRAuthStoredUserManager alloc] initWithServiceName:keychainServiceName];
-      }
+    if ([application isProtectedDataAvailable]) {
```

Comment:
This needs to be gated as only on iOS/tvOS/Catalyst still, but looks like this property was recently added to AppKit: 

https://developer.apple.com/documentation/appkit/nsapplication/3752992-isprotecteddataavailable

Not worrying about the new AppKit API yet, I think the logic should check if this API is available (it was only introduced in Catalyst in 13.1, but 13.0 exists) and if the instance doesn't respond to the selector use the old logic of just trying to read from the keychain.

WDYT?

Reply:
Done -- is there a macro for checking Catalyst version? I see examples of checking iOS version with a macro, but couldn't find one for Catalyst. Or is it enough to have the macro-level checks for the platform, and then also have the `respondsToSelector` check?

Reply:
Not a compile time one - I think checking via `respondsToSelector` on Catalyst makes sense to me.